Regression on ROS 2 Buildfarm for Rolling

`septentrio_gnss_driver 1.4.2` is failing to build on the buildfarm for Rolling. ' The package was likely linking against gtest/ament_cmake_ros transitively but we'll need to be explicit after https://github.com/ros/rosdistro/pull/44971

This is blocking the [announced Rolling sync](https://discourse.ros.org/t/preparing-for-rolling-sync-2025-04-11/43166). We will proceed with the sync which will remove these binaries from Rolling but they will be added once the build is fixed and a new release is bloomed.

See detailed logs: https://build.ros2.org/view/Rbin_uN64/job/Rbin_uN64__septentrio_gnss_driver__ubuntu_noble_amd64__binary/65/console


```
23:15:18 CMake Error at CMakeLists.txt:268 (find_package):
23:15:18   By not providing "Findament_cmake_gtest.cmake" in CMAKE_MODULE_PATH this
23:15:18   project has asked CMake to find a package configuration file provided by
23:15:18   "ament_cmake_gtest", but CMake did not find one.
23:15:18 
23:15:18   Could not find a package configuration file provided by "ament_cmake_gtest"
23:15:18   with any of the following names:
23:15:18 
23:15:18     ament_cmake_gtestConfig.cmake
23:15:18     ament_cmake_gtest-config.cmake
23:15:18 
23:15:18   Add the installation prefix of "ament_cmake_gtest" to CMAKE_PREFIX_PATH or
23:15:18   set "ament_cmake_gtest_DIR" to a directory containing one of the above
23:15:18   files.  If "ament_cmake_gtest" provides a separate development package or
23:15:18   SDK, be sure it has been installed.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,6 +30,8 @@ endif ()
 
 find_package(ament_cmake QUIET)
 find_package(catkin QUIET)
+find_package(gtest_vendor)
+find_package(ament_cmake_ros)
 
 if(catkin_FOUND)
   ## ROS 1 -----------------------------------------------------------------------------------------------  

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,8 +30,6 @@ endif ()
 
 find_package(ament_cmake QUIET)
 find_package(catkin QUIET)
-find_package(gtest_vendor)
-find_package(ament_cmake_ros)
 
 if(catkin_FOUND)
   ## ROS 1 -----------------------------------------------------------------------------------------------  
@@ -163,6 +161,8 @@ elseif(ament_cmake_FOUND)
   find_package(tf2_eigen REQUIRED)
   find_package(tf2_geometry_msgs REQUIRED)
   find_package(tf2_ros REQUIRED)
+  find_package(gtest_vendor REQUIRED)
+  find_package(ament_cmake_ros REQUIRED)
   
   ## Declare  messages
   rosidl_generate_interfaces(${PROJECT_NAME}

--- a/package.xml
+++ b/package.xml
@@ -29,7 +29,9 @@
   <member_of_group condition="$ROS_VERSION == 2">rosidl_interface_packages</member_of_group>
   <depend condition="$ROS_VERSION == 2">rclcpp</depend>
   <depend condition="$ROS_VERSION == 2">rclcpp_components</depend>
-  
+  <depend condition="$ROS_VERSION == 2">ament_cmake_ros</depend>
+  <depend condition="$ROS_VERSION == 2">gtest_vendor</depend>
+
   <buildtool_depend condition="$ROS_VERSION == 1">catkin</buildtool_depend>
   <build_depend condition="$ROS_VERSION == 1">message_generation</build_depend>
   <depend condition="$ROS_VERSION == 1">roscpp</depend>
@@ -49,9 +51,6 @@
   <depend>tf2_eigen</depend>
   <depend>tf2_geometry_msgs</depend>
   <depend>tf2_ros</depend>
-  <depend>ament_cmake_ros</depend>
-  <depend>gtest_vendor</depend>
-
   <export>
     <build_type condition="$ROS_VERSION == 1">catkin</build_type>
     <build_type condition="$ROS_VERSION == 2">ament_cmake</build_type>

--- a/package.xml
+++ b/package.xml
@@ -49,6 +49,8 @@
   <depend>tf2_eigen</depend>
   <depend>tf2_geometry_msgs</depend>
   <depend>tf2_ros</depend>
+  <depend>ament_cmake_ros</depend>
+  <depend>gtest_vendor</depend>
 
   <export>
     <build_type condition="$ROS_VERSION == 1">catkin</build_type>

